spec/queue-spec.js

```diff
@@ -0,0 +1,68 @@
+import Q from 'q'
+import {expect} from 'chai'
+import Queue from '../src/queue.js'
+import Task from '../src/task.js'
+
+class SampleTask extends Task {
+
+  constructor(time) {
+    super()
+    this.time = time
+  }
+
+  run() {
+    const deferred = Q.defer()
+    setTimeout(() => deferred.resolve(), this.time)
+    return deferred.promise
+  }
+
+  done() {
+    // Job done!
+  }
+
+}
+
+describe('Queue', () => {
+
+  it('should add a job to the queue', (done) => {
+    expect(Queue.tasks.length).to.equal(0)
+    Queue.push(new SampleTask(100))
+    expect(Queue.tasks.length).to.equal(1)
+    setTimeout(done, 200)
```

Comment:
Yeah it does because without waiting those 100ms, in the next method this task will still be running.